src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java

```diff
@@ -0,0 +1,197 @@
+    @CheckForNull
+    private Map<String,Object> arguments;
+
+
+    private boolean isModifiedBySanitization = false;
+
+    public boolean isModifiedBySanitization() {
+        return isModifiedBySanitization;
+    }
+
+    public ArgumentsActionImpl(@Nonnull Map<String, Object> stepArguments, @Nullable EnvVars env) {
+        Map<String,Object> sanitizedArguments = sanitizedStepArguments(stepArguments, env);
+        for (Object o : sanitizedArguments.values()) {
+            if (o != null && (o.equals(MASKED_VALUE) || o.equals(NotStoredReason.OVERSIZE_VALUE))) {
+                isModifiedBySanitization = true;
+                break;
+            }
+        }
+        arguments = sanitizedArguments;
+    }
+
+    /** Create a step, sanitizing strings for secured content */
+    public ArgumentsActionImpl(@Nonnull Map<String, Object> stepArguments) {
+        this(stepArguments, new EnvVars());
+    }
+
+    /** See if sensitive environment variable content is in a string */
+    public static boolean isStringSafe(@CheckForNull String input, @CheckForNull EnvVars variables, @Nonnull Set<String> safeEnvVariables) {
+        if (input == null || variables == null || variables.size() == 0) {
+            return true;
+        }
+        StringBuilder pattern = new StringBuilder();
+        int count = 0;
+        for (Map.Entry<String,String> ent : variables.entrySet()) {
+            String val = ent.getValue();
+            if (val == null || val.isEmpty() || safeEnvVariables.contains(ent.getKey())) {  // Skip values that are safe
+                continue;
+            }
+            if (count > 0) {
+                pattern.append('|');
+            }
+            pattern.append(Pattern.quote(val));
+            count++;
+        }
+        return (count > 0)
+                ? !Pattern.compile(pattern.toString()).matcher(input).find()
+                : true;
+    }
+
+    private static final HashSet<String> SAFE_ENVIRONMENT_VARIABLES = new HashSet<String>(Arrays.asList(
+            // Pipeline/Jenkins variables in normal builds
+            "BRANCH_NAME",
+            "BUILD_DISPLAY_NAME",
+            "BUILD_ID",
+            "BUILD_NUMBER",
+            "BUILD_TAG",
+            "BUILD_URL",
+            "CHANGE_AUTHOR",
+            "CHANGE_AUTHOR_DISPLAY_NAME",
+            "CHANGE_AUTHOR_EMAIL",
+            "CHANGE_ID",
+            "CHANGE_TARGET",
+            "CHANGE_TITLE",
+            "CHANGE_URL",
+            "EXECUTOR_NUMBER",
+            "HUDSON_COOKIE",
+            "HUDSON_HOME",
+            "HUDSON_SERVER_COOKIE",
+            "HUDSON_URL",
+            "JENKINS_HOME",
+            "JENKINS_SERVER_COOKIE",
+            "JENKINS_URL",
+            "JOB_BASE_NAME",
+            "JOB_NAME",
+            "JOB_URL",
+            "NODE_LABELS",
+            "NODE_NAME",
+            "WORKSPACE",
+
+            // Normal system variables for posix environments
```

Comment:
Since we're hardcoded-whitelisting, let's try to get the equivalent for Windows land in here too?